pilot/pkg/dns/dns.go

```diff
@@ -354,9 +355,21 @@ func generateAltHosts(hostname string, nameinfo *nds.NameTable_NameInfo, proxyNa
 // of registry, we will look it up in one of our tables, failing which we will return NXDOMAIN.
 func (table *LookupTable) lookupHost(qtype uint16, hostname string) ([]dns.RR, bool) {
 	var hostFound bool
-	if _, hostFound = table.allHosts[hostname]; !hostFound {
-		// this is not from our registry
-		return nil, false
+
+	hname := host.Name(hostname)
+	if hname.IsWildCarded() {
```

Comment:
I donot think this is right, there should be no dns req for wildcarded domain.

https://en.wikipedia.org/wiki/Wildcard_DNS_record#:~:text=A%20wildcard%20DNS%20record%20is,example.com%20.

Reply:
Yes. You are right. The check is not needed.